client/src/pages/Profile.jsx

```diff
@@ -159,7 +224,7 @@ const Profile = () => {
 						className="border min-w-[300px] max-w-[400px] w-[40%] p-2 text-md rounded-xl border-green-500 text-green-500 hover:bg-green-500 hover:text-white dark:border-green-500 dark:text-green-500 dark:hover:bg-green-500 dark:hover:text-white"
 						onClick={handleUpdateProfile}
 					>
-						Save changes
+						{loading? `Processing...` : `Save changes`}
```

Comment:
it can still be clicked on this way, the loading should be above the button, with the second result of the condition should be the button

either that or make it while loading should be unclickable